### PR TITLE
Update serialization how-to

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #20887.

- Update constructor requirements for deserialization.
- Standardize generic types to "WeatherForecast" in snippets.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?branch=pr-en-us-21027#deserialization-behavior).

### CODE REVIEW

File: docs/standard/serialization/snippets/system-text-json-how-to/csharp/RoundtripToString.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public static void Run()
             Console.WriteLine($"Pretty-printed JSON output:\n{jsonString}\n");
 
             // <SnippetDeserialize>
-            weatherForecast = JsonSerializer.Deserialize<WeatherForecastWithPOCOs>(jsonString);
+            weatherForecast = JsonSerializer.Deserialize<WeatherForecast>(jsonString);
```
Comment (gewarren) [inline docs/standard/serialization/snippets/system-text-json-how-to/csharp/RoundtripToString.cs:34]: I updated this because the doc read like this, which was confusing to me:

image

So there's a mismatch between the class name, WeatherForecast, and the type in the generic type parameter (WeatherForecastWithPOCOs).

File: docs/standard/serialization/system-text-json-how-to.md [hunk 2]
```diff
@@ -132,7 +132,9 @@ To deserialize from a string or a file, call the <xref:System.Text.Json.JsonSeri
 
 The following example reads JSON from a string and creates an instance of the `WeatherForecast` class shown earlier for the [serialization example](#serialization-example):
 
-[!code-csharp[](snippets/system-text-json-how-to/csharp/RoundtripToString.cs?name=SnippetDeserialize)]
+```csharp
+WeatherForecast weatherForecast = JsonSerializer.Deserialize<WeatherForecast>(jsonString);
+```
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json-how-to.md:137]: I updated this because the doc read like this, which was confusing to me:

image

So there's a mismatch between the class name, WeatherForecast, and the type in the generic type parameter (WeatherForecastWithPOCOs).

Comment (tdykstra) [reply docs/standard/serialization/system-text-json-how-to.md:137]: The "class shown earlier for the serialization example" is WeatherForecastWithPOCOs, so it might be better here to change the intro sentence rather than the code -- "...creates an instance of the `WeatherForecastWithPOCOs` class shown earlier..."

**Review by tdykstra — APPROVED** [review]
Thanks for these updates - I made one suggestion.

File: docs/standard/serialization/system-text-json-how-to.md [hunk 3]
```diff
@@ -55,17 +55,20 @@ The preceding examples use type inference for the type being serialized. An over
 
 ### Serialization example
 
-Here's an example class that contains collections and a nested class:
+Here's an example class that contains collection-type properties and a user-defined type:
 
 [!code-csharp[](snippets/system-text-json-how-to/csharp/WeatherForecast.cs?name=SnippetWFWithPOCOs)]
 
+> [!TIP]
+> "POCO" stands for [plain old CLR object](https://en.wikipedia.org/wiki/Plain_old_CLR_object). A POCO is a .NET type that isn't derived from a special base class and has no attributes describing infrastructure concerns, for example, property visibility in a designer.
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json-how-to.md:63]: @tdykstra Does this look okay to you?

Comment (tdykstra) [reply docs/standard/serialization/system-text-json-how-to.md:63]: Yes - I think the acronym is common enough that the first sentence with the link would be adequate, but an additional explanatory sentence is fine.  However, "special" base class seems vague. If you keep it, maybe rephrase something like "no base class or attributes describing..."

Comment (gewarren) [inline docs/standard/serialization/system-text-json-how-to.md:63]:
```suggestion
> "POCO" stands for [plain old CLR object](https://en.wikipedia.org/wiki/Plain_old_CLR_object). A POCO is a .NET type that isn't derived from a base class and has no attributes describing infrastructure concerns, for example, property visibility in a designer.
```

File: docs/standard/serialization/system-text-json-how-to.md [hunk 4]
```diff
@@ -60,7 +60,7 @@ Here's an example class that contains collection-type properties and a user-defi
 [!code-csharp[](snippets/system-text-json-how-to/csharp/WeatherForecast.cs?name=SnippetWFWithPOCOs)]
 
 > [!TIP]
-> "POCO" stands for [plain old CLR object](https://en.wikipedia.org/wiki/Plain_old_CLR_object). A POCO is a .NET type that isn't derived from a special base class and has no attributes describing infrastructure concerns, for example, property visibility in a designer.
+> "POCO" stands for [plain old CLR object](https://en.wikipedia.org/wiki/Plain_old_CLR_object). A POCO is a .NET type that isn't derived from a base class and has no attributes describing infrastructure concerns, for example, property visibility in a designer.
```
Comment (tdykstra) [inline docs/standard/serialization/system-text-json-how-to.md:63]: a POCO can be derived from a base class

File: docs/standard/serialization/system-text-json-how-to.md [hunk 5]
```diff
@@ -55,17 +55,20 @@ The preceding examples use type inference for the type being serialized. An over
 
 ### Serialization example
 
-Here's an example class that contains collections and a nested class:
+Here's an example class that contains collection-type properties and a user-defined type:
 
 [!code-csharp[](snippets/system-text-json-how-to/csharp/WeatherForecast.cs?name=SnippetWFWithPOCOs)]
 
+> [!TIP]
+> "POCO" stands for [plain old CLR object](https://en.wikipedia.org/wiki/Plain_old_CLR_object). A POCO is a .NET type that isn't derived from a base class and has no attributes describing infrastructure concerns, for example, property visibility in a designer.
```
Comment (gewarren) [inline docs/standard/serialization/system-text-json-how-to.md:63]:
```suggestion
> "POCO" stands for [plain old CLR object](https://en.wikipedia.org/wiki/Plain_old_CLR_object). A POCO is a .NET type that doesn't depend on any framework-specific types, for example, through inheritance or attributes.
```

**Review by tdykstra — APPROVED** [review]
